pkg/network/tracer/connection/kprobe/manager.go

```diff
@@ -137,3 +162,25 @@ func initManager(mgr *ebpftelemetry.Manager, closedHandler *ebpf.PerfHandler, ru
 
 	return nil
 }
+
+// ComputeDefaultClosedConnRingBufferSize is the default buffer size of the ring buffer for closed connection events.
+// Must be a power of 2 and a multiple of the page size
+func ComputeDefaultClosedConnRingBufferSize() int {
+	numCPU, err := utils.NumCPU()
+	pageSize := os.Getpagesize()
+	if err != nil {
+		numCPU = 1
+	}
+
+	if numCPU <= 16 {
+		return 8 * 8 * pageSize
+	}
```

Comment:
why special behavior for hosts with less than 16 cpus?

Reply:
saw this behavior on the CWS side for process event stream. I removed it for now but I'll ask what the purpose of it was

Reply:
I readded this logic after talking to Paul. ringbuffer size needs to be a power of 2 and ringbuffers are CPU agnostic, so this is a hacky way of scaling the size of the buffer based on cpu since the number of events is generally proportional to that

Reply:
8 - is a power of 2
16 - is a power of 2
8 * 8 - is a power of 2
8 * 16 - is a power of 2

most likely the pagesize is a power of 2 as well (probably 4k, or 8k), but if pagesize * 8 * 8 is a power of 2 then pagesize*8*16 is a power of 2

so I don't understand the different value based on the number of CPUs

Reply:
updated to not use this logic anymore and round the CPU count to the closest power of 2